test/e2e/framework.go

```diff
@@ -57,12 +59,23 @@ type Framework struct {
 func NewFramework(baseName string) *Framework {
 	f := &Framework{
 		BaseName:                 baseName,
-		addonResourceConstraints: make(map[string]resourceConstraint),
+		AddonResourceConstraints: make(map[string]resourceConstraint),
+		NsCreateFunc:             createTestingNS,
 	}
+	return f.InstallGinkgoHooks()
+}
 
+// InstallGinkgoHooks installs the BeforeEach and AfterEach Ginkgo blocks
+func (f *Framework) InstallGinkgoHooks() *Framework {
 	BeforeEach(f.beforeEach)
 	AfterEach(f.afterEach)
+	return f
+}
 
+// SetNsCreateFunc allows to override the default namespace creation function
+// This function must be called before InstallGinkgoHooks() to take effect.
```

Comment:
if you use NewFramework (as we intend everyone to do), there's no way to set this before InstallGinkgoHooks is called.